src/ima.js

```diff
@@ -928,8 +928,15 @@ class Ima extends BasePlugin implements IMiddlewareProvider, IAdsControllerProvi
       if (this._currentAd.isLinear()) {
         this._adsManager.resize(this.player.dimensions.width, this.player.dimensions.height, viewMode);
       } else {
-        this._alignAdsContainerSizeForOverlayAd();
-        this._adsManager.resize(this._currentAd.getWidth() + OVERLAY_AD_MARGIN, this._currentAd.getHeight() + OVERLAY_AD_MARGIN, viewMode);
+        const adTotalWidth = this._currentAd.getWidth() + OVERLAY_AD_MARGIN;
+        const adTotalHeight = this._currentAd.getHeight() + OVERLAY_AD_MARGIN;
+        if (adTotalWidth <= this.player.dimensions.width && adTotalHeight <= this.player.dimensions.height) {
+          this._alignAdsContainerSizeForOverlayAd();
+          this._adsManager.resize(this._currentAd.getWidth() + OVERLAY_AD_MARGIN, this._currentAd.getHeight() + OVERLAY_AD_MARGIN, viewMode);
+          this._showAdsContainer();
+        } else {
+          this._hideAdsContainer();
```

Comment:
if you not hide it and get the error the ad will be displayed in a larger player?

Reply:
After error the ad is down forever...